text/0000-pack-toml.md

```diff
@@ -0,0 +1,123 @@
+# Meta
+[meta]: #meta
+- Name: Support for `pack.toml`
+- Start Date: 2021-11-16
+- Author(s): [jkutner](http://github.com/jkutner)
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a `pack.toml` file that is a special case of the `project.toml` descriptor file. The `pack.toml` will include configuration options specific to the Pack CLI, which may not be honored by other platforms.
+
+# Definitions
+[definitions]: #definitions
+
+- *project descriptor* - a extension spec defining the [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) file.
+
+
+# Motivation
+[motivation]: #motivation
+
+The `project.toml` file has been a useful and essential tool for buildpack users who need to codify certain arguments of their builds. However, there has been much debate around how to handle configuration options that may not be supported by all platforms.
+
+We desire a descriptor file that is:
+* specific to pack (i.e. not platform-neutral)
+* advertises that the repo containing the file can be built with pack
+* adheres to the best practices and learnings from [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md)
+
+# What it is
+[what-it-is]: #what-it-is
+
+The Pack CLI will support a new descriptor file, `pack.toml`, that is a superset of the project descriptor.
+* An app MAY have a `pack.toml`
+* An app MAY have a `project.toml`
+* An app MAY have both a `pack.toml` and a `projec.toml`, in which case they will be merged.
```

Comment:
What is the use case for having both? Wasn't the purpose of https://github.com/buildpacks/rfcs/blob/main/text/0084-project-descriptor-domains.md to allow for different platform-specific fields to exist in a single master file?

I can't help but feeling like the existence of this proposal is a sign that we need to stop and reevaluate `project.toml`. Clearly being to commit config for `pack` is valuable. But requiring all platforms to conform to the same interface isn't working, even for the simplest platforms like pack.

Reply:
I'd be interested in far it seems impossible to get to a common descriptor that all platforms can adhere to? Actually, I'd expect that at least a lot of those could even be provided by a shared implementation. Isn't that what the RFC discussion about a `prepare` lifecycle is about?

If we need platform specific configuration, and `builder` might be such a case as platforms like kpack or project Piper's `cnbBuild` won't read the `project.toml` until well after running in a process on a builder image, platform specific or optionally interpreted keys in a common file sound far superior to separate files that need merging.